packages/gatsby-link/src/index.tsx

```diff
@@ -173,48 +219,53 @@ class GatsbyLink extends React.Component {
             // loaded before continuing.
             navigate(to, { state, replace })
           }
-
-          return true
```

Comment:
Double checking this is an appropriate change. As I understand React doesn't respect any return values (typings also want a void method)